Data Structures - Stacks and Queues/Stacks using Linked List.py

```diff
@@ -19,28 +19,28 @@ def push(self,data):
       self.top = self.bottom
       self.length = 1
     else:
-      self.top.next = new_node
-      self.top = self.top.next
+      new_node.next = self.top
```

Comment:
new_node.next = self.top is wrong because we need self.top -> new_node(and change new_node to self.top) and not the vice versa.

Reply:
self.top -> new_node can be achieved by: new_node.next = self.top; self.top = new_node;

Reply:
Step 1 : All element in stack -> self.top  
Step 2: self.top.next = new_node , All elements in stack -> self.top -> new_node
Step 3: Update new_node as self.top

Reply:
I see why you come up with this answer. But the correct order (based on my understanding as well as the course material) should be the top node points to all the remaining element in stack: self.top --next--> other elements in stack --next--> self.bottom. The result generated by my code also follow this order, by printing the element starts from top to the bottom: apple -> facebook -> microsoft -> google; while your code print: google -> microsoft -> facebook -> apple

Reply:
I see why you come up with this answer. But the correct order (based on my understanding as well as the course material) should be the top node points to all the remaining element in stack: self.top --next--> other elements in stack --next--> self.bottom. The result generated by my code also follow this order, by printing the element starts from top to the bottom: apple -> facebook -> microsoft -> google; while your code print: google -> microsoft -> facebook -> apple

Reply:
Thanks for pointing out, seems like I implemented the other way round..